jsonschema-core.xml

```diff
@@ -581,39 +554,61 @@
-                    (":"), or periods (".").
-                </t>
-                <t>
-                    Providing a plain name fragment enables a subschema to be
-                    relocated within a schema without requiring that JSON
-                    Pointer references are updated.
-                </t>
-                <t>
-                    The effect of defining a fragment-only "$id" URI reference that neither
-                    matches the above requirements nor is a valid JSON pointer
-                    is not defined.
-                </t>
-                <t>
-                    For example:
+                <section title="Identifying the root schema">
+                    <t>
+                        The root schema of a JSON Schema document SHOULD contain an "$id" keyword with
+                        a URI (containing a scheme).  This URI SHOULD either not have a fragment, or
+                        have one that is an empty string.
+                        <!-- All of the standard meta-schemas use an empty fragment in their id/$id values. -->
+                    </t>
+                </section>
+                <section title="Changing the base URI within a schema file">
+                    <t>
+                        When an "$id" sets the base URI, the object containing that "$id" and all of
+                        its subschemas can be identified by using a JSON Pointer fragment starting
+                        from that location.  This is true even of subschemas that further change the
+                        base URI.  Therefore, a single subschema may be accessible by multiple URIs,
+                        each consisting of base URI declared in the subschema or a parent, along with
+                        a JSON Pointer fragment identifying the path from the schema object that
+                        declares the base to the subschema being identified.  Examples of this are
+                        shown in section <xref target="idExamples" format="counter"></xref>.
+                    </t>
+                </section>
+                <section title="Location-independent identifiers">
+                    <t>
+                        Using JSON Pointer fragments requires knowledge of the structure of the schema.
+                        When writing schema documents with the intention to provide re-usable
+                        schemas, it is preferable to use a plain name fragment that is not tied to
+                        any particular structural location.  This allows a subschema to be relocated
+                        without requiring JSON Pointer references to be updated.
+                    </t>
+                    <t>
+                        To name subschemas in a JSON Schema document,
+                        subschemas can use "$id" to give themselves a document-local identifier.
```

Comment:
Rereading this language I know I wrote, it may be clearer to add a qualifier to '"$id"' like

> To name subschemas in a JSON Schema document,
> subschemas can use a bare fragment for "$id" to give themselves a document-local identifier.

Maybe I can PR for this after merge.